dlib/dnn/loss.h

```diff
@@ -1140,6 +1139,7 @@ namespace dlib
                     det_thresh_speed_adjust = std::max(det_thresh_speed_adjust,dets[max_num_initial_dets].detection_confidence + options.loss_per_false_alarm);
                 }
 
+                std::vector<size_t> truth_idxs;  truth_idxs.reserve(truth->size());
```

Comment:
And make this two lines.  Not sure what I was thinking with this funky formatting